central/role/datastore/datastore_impl.go

```diff
@@ -77,7 +77,7 @@ func (ds *dataStoreImpl) AddRole(ctx context.Context, role *storage.Role) error
 	if err := rolePkg.ValidateRole(role); err != nil {
 		return errors.Wrap(errox.InvalidArgs, err.Error())
 	}
-	if err := verifyNotDefaultRole(role.GetName()); err != nil {
+	if err := verifyNotDefaultRole(role); err != nil {
```

Comment:
I wonder if there is a benefit of having this check in `AddRole` method considering adding role with default name should be impossible just because the name is occupied.

Reply:
I think it's better to be explicit here, as `verifyNotDefaultRole` returns `InvalidArgs` with a specific error message that default roles cannot be added / modified / deleted, compared to already exists.

I'm also quite sure this is a breaking change in our API, since we would return 400 instead of 409, which we return for "AlreadyExists".